### PR TITLE
[KYUUBI #3653][REST] AdminResource add list kyuubi server api

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Add List Kyuubi Server Api for `AdminResource`

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zwangsheng** [comment]
cc @turboFei @pan3793

**Comment by turboFei** [comment]
some comments:
1. we can support `discoveryService` for KyuubiRestFrontendService
2. we shall return the rest url for the API in this pr

**Comment by zwangsheng** [comment]
> some comments:
> 
> 1. we can support `discoveryService` for KyuubiRestFrontendService
> 2. we shall return the rest fe urls for the API in this pr

@turboFei 

Sure, made changes.

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/AdminResourceSuite.scala [hunk 1]
```diff
@@ -488,4 +491,41 @@ class AdminResourceSuite extends KyuubiFunSuite with RestFrontendTestHelper {
...
+    lenient.when(serverDiscovery.fe).thenReturn(fe)
+    val namespace = conf.get(HighAvailabilityConf.HA_NAMESPACE)
+    withDiscoveryClient(conf) { client =>
+      client.registerService(conf, namespace, serverDiscovery)
+
+      val adminUser = Utils.currentUser
+      val encodeAuthorization = new String(
+        Base64.getEncoder.encode(
+          s"$adminUser:".getBytes()),
+        "UTF-8")
```
Comment (bowenliang123) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/AdminResourceSuite.scala:507]:
```suggestion
        StandardCharsets.UTF_8)
```

Comment (zwangsheng) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/AdminResourceSuite.scala]: Thanks for point this.

By the way, other similar places in this document have been modified and duplicate code has been reduced.

**Comment by zwangsheng** [comment]
Also cc @pan3793

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala [hunk 2]
```diff
@@ -296,6 +296,35 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
...
+    val ipAddress = fe.getIpAddress
+    info(s"Received list all live kyuubi servers request from $userName/$ipAddress")
+    if (!isAdministrator(userName)) {
+      throw new NotAllowedException(
+        s"$userName is not allowed to list all live kyuubi servers")
+    }
+    val kyuubiConf = fe.getConf
+    val servers = ListBuffer[ServerData]()
+    val serverSpec = DiscoveryPaths.makePath(null, kyuubiConf.get(HA_NAMESPACE))
+    withDiscoveryClient(kyuubiConf) { discoveryClient =>
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:320]: what will happen if HA is disabled

Comment (zwangsheng) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:320]: When disabled HA, `Kyuubi Server` will register self to `EmbeddedZookeeper`.

It's predictably to get only itself when disabled HA.

**Review by pan3793 — APPROVED** [review]

**Review by turboFei — APPROVED** [review]
LGTM,

BTW, I recommend to support discovery service for KyuubiRestFrontendService.

And the URLs return in the api added in this PR should be the connection urls of `KyuubiRestFrontendService`s.

**Comment by zwangsheng** [comment]
> BTW, I recommend to support discovery service for KyuubiRestFrontendService.
> And the URLs return in the api added in this PR should be the connection urls of `KyuubiRestFrontendService`s.

I agree with support discovery service for `KyuubiRestFrontendService` and  expose rest connection url in HA node.

Seems this is a different goal feature, i'd like make a new PR to implement.

**Comment by zwangsheng** [comment]
ping @pan3793 @turboFei, we can merge this.

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**